runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 			MenuEntry[] menuEntries = client.getMenuEntries();
 			MenuEntry lastEntry = menuEntries[menuEntries.length - 1];
 
-			int quantity = 1;
-			Node current = itemLayer.getBottom();
-
-			while (current instanceof TileItem)
-			{
-				TileItem item = (TileItem) current;
-				if (item.getId() == itemId)
-				{
-					quantity = item.getQuantity();
-				}
-				current = current.getNext();
-			}
+			GroundItem.GroundItemKey groundItemKey = new GroundItem.GroundItemKey(itemId, tile.getWorldLocation());
+			GroundItem groundItem = collectedGroundItems.get(groundItemKey);
+			int quantity = groundItem.getQuantity();
 
-			final ItemComposition itemComposition = itemManager.getItemComposition(itemId);
-			final int realItemId = itemComposition.getNote() != -1 ? itemComposition.getLinkedNoteId() : itemComposition.getId();
-			final int itemPrice = itemManager.getItemPrice(realItemId);
-			final int price = itemPrice <= 0 ? itemComposition.getPrice() : itemPrice;
-			final int haPrice = Math.round(itemComposition.getPrice() * Constants.HIGH_ALCHEMY_MULTIPLIER) * quantity;
-			final int gePrice = quantity * price;
-			final Color hidden = getHidden(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice, itemComposition.isTradeable());
-			final Color highlighted = getHighlighted(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice);
+			final int itemPrice = groundItem.getGePrice();
+			final int haPrice = groundItem.getHaPrice();
+			final int gePrice = itemPrice <= 0 ? haPrice : itemPrice;
```

Comment:
Why is this here? The overlay doesn't have that:

```java
		for (GroundItem item : groundItemList)
		{
			final LocalPoint groundPoint = LocalPoint.fromWorld(client, item.getLocation());

			if (groundPoint == null || localLocation.distanceTo(groundPoint) > MAX_DISTANCE
				|| (onlyShowLoot && !item.isMine()))
			{
				continue;
			}

			final Color highlighted = plugin.getHighlighted(new NamedQuantity(item), item.getGePrice(), item.getHaPrice());
```

Reply:
I assume you mean the ternary operator that uses the HA price when GE is 0? If not please clarify what you mean by "this".
I tried to preserve the existing logic as best as I could. It's possible that this can be removed, I'll test it and report back.

Please compare with the version from branch master:
https://github.com/runelite/runelite/blob/1c9c71ef7c93ccf4c479b7137635e5e266d07788/runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java#L514

Reply:
This price defaulting is indeed not necessary. I removed it.